src/item.rs

```diff
@@ -485,23 +420,130 @@ pub fn is_page_empty<S: NorFlash>(
+                .read_item(flash, data_buffer, address, self.header.end_address)
+                .await?
+            {
+                MaybeItem::Corrupted { .. } => {
+                    #[cfg(feature = "defmt")]
+                    defmt::error!(
+                        "Found a corrupted item at {:X}. Skipping...",
+                        self.header.current_address
+                    );
+                    continue;
+                }
+                MaybeItem::Erased(_) => {}
+                MaybeItem::Present(item) => {
+                    return Ok(Some((item, address)));
+                }
+            }
+        }
+        Ok(None)
+    }
+}
+
+pub struct HeaderIter {
+    current_address: u32,
+    end_address: u32,
+}
+
+impl HeaderIter {
+    pub fn new(start_address: u32, end_address: u32) -> Self {
+        Self {
+            current_address: start_address,
+            end_address,
+        }
+    }
+
+    /// Fetch next item
+    pub async fn next<S: NorFlash>(
+        &mut self,
+        flash: &mut S,
+    ) -> Result<(Option<ItemHeader>, u32), Error<S::Error>> {
+        self.traverse(flash, |_, _| ControlFlow::Break(())).await
+    }
+
+    /// Traverse all headers until the callback determines the next element.
+    ///
+    /// If the end of the headers is reached, a `None` item header is returned.
+    pub async fn traverse<S: NorFlash>(
+        &mut self,
+        flash: &mut S,
+        callback: impl Fn(&ItemHeader, u32) -> ControlFlow<(), ()>,
+    ) -> Result<(Option<ItemHeader>, u32), Error<S::Error>> {
+        if self.current_address >= self.end_address {
+            return Ok((None, self.current_address));
+        }
+
+        loop {
+            match ItemHeader::read_new(flash, self.current_address, self.end_address).await {
+                Ok(Some(header)) => {
+                    let next_address = header.next_item_address::<S>(self.current_address);
+                    match callback(&header, self.current_address) {
+                        ControlFlow::Continue(_) => {
+                            self.current_address = next_address;
+                            continue;
+                        }
+                        ControlFlow::Break(_) => {
+                            let current_address = self.current_address;
+                            self.current_address = next_address;
+                            return Ok((Some(header), current_address));
+                        }
+                    }
+                }
+                Ok(None) => {
+                    return Ok((None, self.current_address));
+                }
+                Err(Error::Corrupted { .. }) => {
+                    #[cfg(feature = "defmt")]
+                    defmt::error!(
+                        "Found a corrupted item header at {:X}. Skipping...",
+                        self.current_address
+                    );
+                    self.current_address += S::WORD_SIZE as u32;
```

Comment:
This branch was changed in #16 

```suggestion
                    self.current_address = ItemHeader::data_address::<S>(self.current_address);
```